dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,323 @@
+package com.datadog.gradle.plugin
+
+import java.io.File
+import java.util.Properties
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.gradle.testkit.runner.UnexpectedBuildFailure
+import org.gradle.testkit.runner.internal.PluginUnderTestMetadataReading
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+
+internal class DdAndroidGradlePluginFunctionalTest {
+    @TempDir
+    lateinit var testProjectDir: File
+    lateinit var mainSrcDir: File
+    lateinit var settingsFile: File
+    lateinit var buildFile: File
+    lateinit var manifestFile: File
+    lateinit var gradlePropertiesFile: File
+
+    @BeforeEach
+    fun `set up`() {
+        mainSrcDir = File(testProjectDir, "src/main").apply { mkdirs() }
+        settingsFile = File(testProjectDir, "settings.gradle")
+        buildFile = File(testProjectDir, "build.gradle")
+        gradlePropertiesFile = File(testProjectDir, "gradle.properties")
+        manifestFile = File(mainSrcDir, "AndroidManifest.xml")
+        stubFile(settingsFile, SETTINGS_FILE_CONTENT)
+        stubFile(manifestFile, ANDROID_MANIFEST_FILE_CONTENT)
+        stubFile(gradlePropertiesFile, GRADLE_PROPERTIES_FILE_CONTENT)
+    }
+
+    // region Assemble
+
+    @Test
+    fun `M success W assemble`() {
```

Comment:
can we try to add the test (well, project structure with app module with flavors + lib module without flavors) which will check that regression of this bug https://github.com/DataDog/dd-sdk-android-gradle-plugin/issues/62 doesn't happen?

Reply:
Ok, I will try to see how we can achieve this.